core/src/robot/AbstractTransmission.cc

```diff
@@ -0,0 +1,365 @@
+#include "jiminy/core/robot/Robot.h"
+#include "jiminy/core/Macros.h"
+
+#include "jiminy/core/utilities/Pinocchio.h"
+#include "jiminy/core/robot/AbstractTransmission.h"
+
+
+namespace jiminy
+{
+    AbstractTransmissionBase::AbstractTransmissionBase(std::string const & name) :
+    baseTransmissionOptions_(nullptr),
+    transmissionOptionsHolder_(),
+    isInitialized_(false),
+    isAttached_(false),
+    robot_(),
+    name_(name),
+    transmissionIdx_(-1),
+    jointNames_(),
+    jointModelIndices_(),
+    jointTypes_(),
+    jointPositionIndices_(),
+    jointVelocityIndices_(),
+    motorNames_(),
+    motorIndices_(),
+    motors_(),
+    forwardJacobian_(),
+    backwardJacobian_()
+    {
+        // TODO initialize the options
+    }
+
+    AbstractTransmissionBase::~AbstractTransmissionBase(void)
+    {
+        // Detach the transmission before deleting it if necessary
+        if (isAttached_)
+        {
+            detach();
+        }
+    }
+
+    hresult_t AbstractTransmissionBase::initialize(std::vector<std::string> const & jointNames,
+                                                   std::vector<std::string> const & motorNames)
+    {
+        // Check whether transmission is invertible or not
+        auto robot = robot_.lock();
+        unsigned int nDofs = 0;
+        jointIndex_t jointModelIdx;
+        for (std::string jointName : jointNames)
+        {
+            ::jiminy::getJointModelIdx(robot->pncModel_, jointName, jointModelIdx);
+            nDofs += robot->pncModel_.joints[jointModelIdx].nv();
+        }
+        if (nDofs != motorNames.size())
+        {
+            PRINT_ERROR("No 1-to-1 mapping from motors to joints");
+            return hresult_t::ERROR_INIT_FAILED;
```

Comment:
only early return if you can early return everywhere in the method (mainly you never have to manage `returnCode` variable directly). Otherwise, use `if (returnCode != hresult_t::SUCCESS)` everywhere.